### PR TITLE
Extra examples

### PR DESCRIPTION
**Pull request by grctest**
Work in progress

Related to: https://github.com/bitshares/beet/pull/207

### CODE REVIEW

**Comment by grctest** [comment]
Operations 32, 34, 36, 69 and 71 need a peer review, I wasn't sure what the input format was supposed to be like where marked `// TODO: DEBUG`

File: examples/bitshares/injections/034_worker_create.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "wss://testnet.xbts.io/ws", // wss url
+    "worker_create", // operation name
+    {
+      owner: "1.2.",
+      work_begin_date: 1663766042,
+      work_end_date: 2663766042,
+      daily_pay: 1,
+      name: "",
+      url: "",
+      initializer: "burn_worker_initializer" // TODO: DEBUG
```
Comment (abitmore) [inline examples/bitshares/injections/034_worker_create.js:16]: This is a static variant too, E.G. `"initializer": [ 0,{} ]` or
```
"initializer": [
              1,{
                "pay_vesting_period_days": 1
              }
            ]
```

File: examples/bitshares/injections/036_assert.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+let run = async function () { 
+  await injection(
+    "InjectExample.036", // script name
+    "BTS_TEST", // chain
+    "wss://testnet.xbts.io/ws", // wss url
+    "assert", // operation name
+    {
+      fee_paying_account: "1.2.",
+      predicates: [ // TODO: DEBUG
```
Comment (abitmore) [inline examples/bitshares/injections/036_assert.js:11]: This is a list of static variants. Format example
```
"predicates": [
  [0, {"account_id":"1.2.1","name":"abc"}],
  [1, {"asset_id":"1.3.2","symbol":"ABC"}]
]
```

**Review by abitmore — COMMENTED** [review]
Thanks!

File: examples/bitshares/injections/074_credit_deal_expired.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { injection } from '../../lib/injection.js'
+
+let run = async function () { 
+  await injection(
+    "InjectExample.074", // script name
```
[the remark this answers is not in the document]
Comment (abitmore) [reply examples/bitshares/injections/074_credit_deal_expired.js]: More info in https://github.com/bitshares/bitshares-core/blob/master/libraries/protocol/include/graphene/protocol/operations.hpp

The virtual operations are
* `/*  4 */ fill_order_operation,           // VIRTUAL`
* `/* 42 */ asset_settle_cancel_operation,  // VIRTUAL`
* `/* 44 */ fba_distribute_operation,       // VIRTUAL`
* `/* 46 */ execute_bid_operation,          // VIRTUAL`
* `/* 51 */ htlc_redeemed_operation,         // VIRTUAL`
* `/* 53 */ htlc_refund_operation,           // VIRTUAL`
* `/* 74 */ credit_deal_expired_operation    // VIRTUAL`

File: examples/lib/injection.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import { connect, link } from '../../../src/index.js';
```
Comment (sschiessl-bcp) [inline examples/lib/injection.js:1]: this seems like too many folders up?

**Comment by sschiessl-bcp** [comment]
With the current beet rc I get stuck doing this

```
>> node ./bitshares/injections/001_limit_order_create.js 
received connected socket response
socket: authenticated
Connected to Beet
sending link request
sending linkRequest request
sending api request
```

No popup visible. Which Beet version are you using?

**Comment by grctest** [comment]
Sorry for the delay, working on the reported issue

**Comment by grctest** [comment]
@sschiessl-bcp Please use the following Beet PR: https://github.com/bitshares/beet/pull/222

I've introduced several fixes to address the issue you ran into, including:
* Tweaks to establishing blockchain connection when switching between blockchains BTS<->BTS_TEST
* Timeouts have been introduced to several functions
* More verbose error checks to reject erroneous injected parameters

More work could be done on the Bitshares optype visualize function, ie returning null if required external data fails to fetch, rejecting the prompt in a safe manner. Could do a follow up PR in the future to improve this function.

TLDR I too wasn't getting the prompt, now I do get it in both production and testnet. If the prompt still doesn't show, try different asset ids etc?

**Comment by sschiessl-bcp** [comment]
see https://github.com/bitshares/beet/pull/222
